Entry for Swift Comparable protocol (#2988)
Codecademy/docs: content/swift/concepts/protocols/terms/comparable/comparable.md

---
Title: 'Comparable Protocol' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
Description: 'Protocol in Swift used to define a sort order for instances of a type.' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
  - 'Mobile Development'
  - 'Computer Science'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
  - 'iOS'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **`Comparable`** protocol in Swift is used to define a sort order for instances of a type. To add `Comparable` conformance to custom types, define the < and == operators as static methods of those types.

Reviewer: ```suggestion
The **`Comparable`** protocol in Swift is used to define a sort order for instances of a type. To add `Comparable` conformance to custom types, define the `<` and `==` operators as static methods of those types.
```

> ## Syntax

Reviewer: add an empty line after the header

```
struct StructName: Comparable {
    let var: Int

    static func < (lhs: StructName, rhs: StructName) -> Bool {
    return lhs.var < rhs.var
    }

    static func == (lhs: StructName, rhs: StructName) -> Bool {
    return lhs.var == rhs.var
    }
}

> ```

Reviewer: add an empty line after the code block

> To conform a custom type to the **Comparable** protocol conformance to **Comparable** is first declared, then static methods for < and == are defined.

Reviewer: ```suggestion
To conform a custom type to the `Comparable` protocol conformance to `Comparable` is first declared, then static methods for `<` and `==` are defined.
```

## Example

```

> struct Building: Comparable {

Reviewer: the code block should use 2 spaces, not tabs, for indentations

    let height: Int
  
    static func < (lhs: Building, rhs: Building) -> Bool {
    return lhs.height < rhs.height
    }

    static func == (lhs: Building, rhs: Building) -> Bool {
    return lhs.height == rhs.height
    }
}

let eiffelTower = Building(height: 300)
let burjKhalifa = Building(height: 828)
let sydneyTower = Building(height: 300)
```

> A custom type `Building` was created with a `height` constant, and static methods for < and == were defined. There are three instances of this type: `eiffelTower`, `BurjKhalifa` and `sydneyTower`. The example conforms to the `Comparable` protocol, allowing each of the instances to be compared:

Reviewer: ```suggestion
A custom type `Building` was created with a `height` constant, and static methods for `<` and `==` were defined. There are three instances of this type: `eiffelTower`, `burjKhalifa` and `sydneyTower`. The example conforms to the `Comparable` protocol, allowing each of the instances to be compared:
```

Reviewer: add an empty line between the code and this section

---
Title: 'Comparable Protocol' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed

> Description: 'Protocol in Swift used to define a sort order for instances of a type' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)

Reviewer 2: ```suggestion
Description: 'Protocol in Swift used to define a sort order for instances of a type.' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
```

Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
  - 'Mobile Development'
  - 'Computer Science'
Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!

>   - 'Protocols'

Reviewer 2: This tag is not in the list here: https://github.com/Codecademy/docs/blob/main/documentation/tags.md

Please use tags from that list.

Author: Changed tag to 'iOS'

> 

Reviewer 2: ```suggestion
```
Delete extra line

CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **Comparable** protocol in Swift is used to define a sort order for instances of a type.
> > To add Comparable conformance to your custom types, define the < and == operators as static methods of your types.

Reviewer 2: ```suggestion
The **`Comparable`** protocol in Swift is used to define a sort order for instances of a type. To add `Comparable` conformance to custom types, define the < and == operators as static methods of those types.
```

## Syntax

> 
> 
> 

Reviewer 2: ```suggestion

```
Only one blank line.

```
struct StructName: Comparable {
    let var: Int

    static func < (lhs: StructName, rhs: StructName) -> Bool {
    return lhs.var < rhs.var
    }

>     static func == (lhs: Building, rhs: Building) -> Bool {

Reviewer 2: ```suggestion
    static func == (lhs: StructName, rhs: StructName) -> Bool {
```

    return lhs.var == rhs.var
    }
}
```

> 

Reviewer 2: We generally have a sentence or two describing what the syntax is showing.

Author: Added sentences describing the syntax.

## Example

> 

Reviewer 2: We generally have a sentence introducing the example code and summarizing what it is doing.

Author: Added sentences describing the example. Thank you for reviewing this entry. I have made all of your suggested changes.
